core/kotlinx-coroutines-core/src/internal/MainDispatchers.kt

```diff
@@ -12,9 +12,12 @@ internal object MainDispatcherLoader {
     private fun loadMainDispatcher(): MainCoroutineDispatcher {
         return try {
             val factories = MainDispatcherFactory::class.java.let { clz ->
-                ServiceLoader.load(clz, clz.classLoader).toList()
+                try {
+                    loadProviders(clz, clz.classLoader).toList()
+                } catch (e: ServiceConfigurationError) {
+                    ServiceLoader.load(clz, clz.classLoader).toList()
```

Comment:
This logic is better to integrate into our service loader.
Also, killswitch is missing (e.g. see `Debug#RECOVER_STACKTRACES`)